tests/integration/dump/test_dump_dbms.py

```diff
@@ -149,15 +152,33 @@ def error_callback(resp):
 
     dump_path = os.path.join(str(tmpdir), 'dump_dir')
     os.environ['DBMS_WORKSPACE'] = os.path.join(str(tmpdir), 'index_ws')
+    os.environ['QUERY_WORKSPACE'] = os.path.join(str(tmpdir), 'query_ws')
+
     with Flow.load_config('flow_dbms.yml') as flow_dbms:
-        with TimeContext(f'### indexing {len(docs)} docs'):
-            flow_dbms.index(docs)
+        with Flow.load_config('flow_query.yml') as flow_query:
+            with TimeContext(f'### indexing {len(docs)} docs'):
+                flow_dbms.index(docs)
+
+            with TimeContext(f'### dumping {len(docs)} docs'):
+                flow_dbms.dump('indexer_dbms', dump_path, shards=shards, timeout=-1)
+
+            dir_size = path_size(dump_path)
+            print(f'### dump path size: {dir_size} MBs')
 
-        with TimeContext(f'### dumping {len(docs)} docs'):
-            flow_dbms.dump('indexer_dbms', dump_path, shards=shards, timeout=-1)
+            with TimeContext(f'### rolling update on {len(docs)}'):
+                flow_query.rolling_update('indexer_query', dump_path)
+
+            # TODO perform query
+            flow_query.search(
+                docs[:nr_search],
+                on_done=_validate_results_nonempty,
+                on_error=error_callback,
+            )
 
-        dir_size = path_size(dump_path)
-        print(f'### dump path size: {dir_size} MBs')
+            # TODO dump again new data
+            # repeat rolling update search
+            # TODO maybe async
```

Comment:
It has to be done separating the processes where the `client` sends data and the main processes where the `flow` objects exist. Otherwise we can't test that we do query_while_indexing in a non-blocking way

Reply:
Yes, good point. I will write it in that approach